app/src/main/scala/org/alephium/explorer/service/BlockFlowClient.scala

```diff
@@ -167,13 +167,80 @@ object BlockFlowClient {
     }
   }
 
+  def blockProtocolToInputEntities(block: api.model.BlockEntry): Seq[InputEntity] = {
+    val hash         = new BlockEntry.Hash(block.hash)
+    val mainChain    = false
+    val transactions = block.transactions.toSeq.zipWithIndex
+    val inputs =
+      transactions.flatMap {
+        case (tx, txIndex) =>
+          tx.unsigned.inputs.toSeq.zipWithIndex.map {
+            case (in, index) =>
+              inputToEntity(in, hash, tx.unsigned.hash, block.timestamp, mainChain, index, txIndex)
+          }
+      }
+    val contractInputs =
+      transactions.flatMap {
+        case (tx, txIndex) =>
+          tx.contractInputs.toSeq.zipWithIndex.map {
+            case (outputRef, index) =>
+              val shiftIndex = index + inputs.size
```

Comment:
should be `index + tx.unsigned.inputs.size` instead of `inputs.size`?

Reply:
oooh nice catch, I used `inputs` thinking it's the same, but inputs is actually `transactions.flatMap(_.unsigned.inputs.size)`